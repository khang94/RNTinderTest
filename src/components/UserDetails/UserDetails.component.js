import React from 'react';
import {Text, View, TouchableOpacity, Image} from 'react-native';
import PropTypes from 'prop-types';

import styles from './UserDetails.styles';
import assets from './../../assets/images';

const UserDetails = ({title, content, activeTab, onSelectItem}) => (
  <View style={styles.container}>
    <View style={styles.info}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.content}>{content}</Text>
    </View>
    <View style={styles.iconList}>
      <TouchableOpacity onPress={() => onSelectItem('account')}>
        {activeTab === 'account' ? (
          <Image source={assets.accountActive} style={styles.icon} />
        ) : (
          <Image source={assets.account} style={styles.icon} />
        )}
      </TouchableOpacity>
      <TouchableOpacity onPress={() => onSelectItem('birthday')}>
        {activeTab === 'birthday' ? (
          <Image source={assets.birthdayActive} style={styles.icon} />
        ) : (
          <Image source={assets.birthday} style={styles.icon} />
        )}
      </TouchableOpacity>
      <TouchableOpacity onPress={() => onSelectItem('location')}>
        {activeTab === 'location' ? (
          <Image source={assets.locationActive} style={styles.icon} />
        ) : (
          <Image source={assets.location} style={styles.icon} />
        )}
      </TouchableOpacity>
      <TouchableOpacity onPress={() => onSelectItem('phone')}>
        {activeTab === 'phone' ? (
          <Image source={assets.phoneActive} style={styles.icon} />
        ) : (
          <Image source={assets.phone} style={styles.icon} />
        )}
      </TouchableOpacity>
      <TouchableOpacity onPress={() => onSelectItem('password')}>
        {activeTab === 'password' ? (
          <Image source={assets.passwordActive} style={styles.icon} />
        ) : (
          <Image source={assets.password} style={styles.icon} />
        )}
      </TouchableOpacity>
    </View>
  </View>
);

UserDetails.defaultProps = {
  title: '',
  content: '',
  activeTab: '',
  onSelectItem: () => {},
};

UserDetails.propTypes = {
  activeTab: PropTypes.string.isRequired,
  onSelectItem: PropTypes.func.isRequired,
};

export default UserDetails;
