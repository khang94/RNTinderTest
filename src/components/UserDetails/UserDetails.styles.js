import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
  },
  iconList: {
    paddingTop: 30,
    flexDirection: 'row',
  },
  icon: {
    width: 50,
    height: 60,
  },
  info: {
    height: 80,
    alignItems: 'center',
    paddingVertical: 20,
  },
  title: {
    fontSize: 18,
    color: '#BDBDBD',
  },
  content: {
    fontSize: 23,
    color: 'black',
  },
});
