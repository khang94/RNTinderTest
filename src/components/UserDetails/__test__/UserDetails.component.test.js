import React from 'react';

import UserDetails from '../UserDetails.component';
import {shallow} from 'enzyme';

describe('UserDetails snapshot', () => {
  const props = {
    title: 'title',
    content: 'content',
    activeTab: 'home',
    onSelectItem: jest.fn(),
  };

  it('Should render snapshot for user details view', () => {
    const wrapper = shallow(<UserDetails {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render active location icon', () => {
    const componentProps = {
      ...props,
      activeTab: 'location',
    };
    const wrapper = shallow(<UserDetails {...componentProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Should render active birthday icon', () => {
    const componentProps = {
      ...props,
      activeTab: 'birthday',
    };
    const wrapper = shallow(<UserDetails {...componentProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Should render active email icon', () => {
    const componentProps = {
      ...props,
      activeTab: 'email',
    };
    const wrapper = shallow(<UserDetails {...componentProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Should render active password icon', () => {
    const componentProps = {
      ...props,
      activeTab: 'password',
    };
    const wrapper = shallow(<UserDetails {...componentProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Should render active phone icon', () => {
    const componentProps = {
      ...props,
      activeTab: 'phone',
    };
    const wrapper = shallow(<UserDetails {...componentProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Should render active account icon', () => {
    const componentProps = {
      ...props,
      activeTab: 'account',
    };
    const wrapper = shallow(<UserDetails {...componentProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
