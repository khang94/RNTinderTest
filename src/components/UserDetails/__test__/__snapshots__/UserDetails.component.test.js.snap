// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails snapshot Should render active account icon 1`] = `ShallowWrapper {}`;

exports[`UserDetails snapshot Should render active birthday icon 1`] = `ShallowWrapper {}`;

exports[`UserDetails snapshot Should render active email icon 1`] = `ShallowWrapper {}`;

exports[`UserDetails snapshot Should render active location icon 1`] = `ShallowWrapper {}`;

exports[`UserDetails snapshot Should render active password icon 1`] = `ShallowWrapper {}`;

exports[`UserDetails snapshot Should render active phone icon 1`] = `ShallowWrapper {}`;

exports[`UserDetails snapshot Should render snapshot for user details view 1`] = `ShallowWrapper {}`;
